java/mdsplus-api/src/main/java/mds/mdsip/MdsIp.java

```diff
@@ -438,7 +433,7 @@ synchronized private final void connectToServer() throws IOException {
 		message.useCompression(this.use_compression);
 		long tictoc = -System.nanoTime();
 		message.send(this.connection);
-		final Message msg = Message.receive(this.connection, null);
+		final Message msg = Message.receive(this.connection, null, 3000);
```

Comment:
Should this 3000 be hardcoded?

Reply:
well for now i think this suffices.. the api does not have a timeout value. on chould think of adding timeout as argumenr to connectoToServer

Reply:
That should probably be a TODO, but yeah that should suffice